### ISSUE TITLE
SCPError vs SFTPError

### ISSUE THREAD

**Issue by termim** [ronf/asyncssh#273]
The *SFTPError* exception is usable as *asyncssh.SFTPError* while for *SCPError* the *asyncssh.scp.SCPError* is needed. May be it makes sense to make *SCPError* top level as well?

**Comment by ronf** [comment]
No - this is intentional. The SCPError class is intended to be internal to the scp module. It adds some additional members that are needed to properly return errors in the form needed by the SCP wire protocol. However, SCPError is a subclass of SFTPError, and that's what application code is expected to catch. In the case of the server-side implementation, SCP and SFTP actually both rely on the same SFTPServer object, which should only ever raise SFTPError, not SCPError.

That said, I failed to update the SCP implementation to properly take into account the recent changes I made which introduced subclasses of SFTPError that applications can catch. I'll remedy that before I put out the next release.

**Comment by ronf** [comment]
Ok - this should be corrected in commit abc87f7 in the "develop" branch.

While it was never intended to be public, I never really liked the way SCPError showed up in debugging output, as that could be confusing. So, as part of this change, I decided to eliminate SCPError entirely and directly create one of the public SFTPError subclasses when raising SCP errors. This also had the benefit of my not having to deal with multiple inheritance. The SCP code already had to cope with exceptions not always having the extra "fatal" and "suppress_send" members, since they weren't always of type SCPError. Now, I've just eliminated that type and add these members directly to SFTPError instances in cases where they're needed.
